nilearn/input_data/base_masker.py

```diff
@@ -5,75 +5,85 @@
+        class_name = enclosing_scope_name(stack_level=3)
+
     # If we have a string (filename), we won't need to copy, as
     # there will be no side effect
-
     if isinstance(imgs, _basestring):
         copy = False
 
     if verbose > 0:
-        class_name = enclosing_scope_name(stack_level=2)
-
-    mask_img_ = _utils.check_niimg_3d(mask_img_)
-
+        print("[%s] Loading data from %s" % (
+            class_name,
+            _utils._repr_niimgs(imgs)[:200]))
     imgs = _utils.check_niimg(imgs, atleast_4d=True, ensure_ndim=4)
+
+    sample_mask = parameters.get('sample_mask', None)
     if sample_mask is not None:
         imgs = image.index_img(imgs, sample_mask)
 
-    # Resampling: allows the user to change the affine, the shape or both
-    if verbose > 1:
-        print("[%s] Resampling" % class_name)
-
-    # Check whether resampling is truly necessary. If so, crop mask
-    # as small as possible in order to speed up the process
-    if not _check_same_fov(imgs, mask_img_):
-        # now we can crop
-        mask_img_ = image.crop_img(mask_img_, copy=False)
+    target_shape = parameters.get('target_shape', None)
+    target_affine = parameters.get('target_affine', None)
+    if target_shape is not None or target_affine is not None:
+        if verbose > 0:
+            print("[%s] Resampling images" % class_name)
+        imgs = cache(
+            image.resample_img, memory, func_memory_level=2,
+            memory_level=memory_level, ignore=['copy'])(
+                imgs, interpolation="continuous",
+                target_shape=target_shape,
+                target_affine=target_affine,
+                copy=copy)
 
-        imgs = cache(image.resample_img, memory, func_memory_level=2,
-                     memory_level=memory_level, ignore=['copy'])(
-                        imgs,
-                        target_affine=mask_img_.get_affine(),
-                        target_shape=mask_img_.shape,
-                        copy=copy)
+    if 'smoothing_fwhm' in parameters:
+        if verbose > 0:
+            print("[%s] Smoothing images" % class_name)
+        imgs = cache(
+            image.smooth_img, memory, func_memory_level=2,
+            memory_level=memory_level)(
+                imgs, parameters['smoothing_fwhm'])
 
-    # Load data (if filenames are given, load them)
     if verbose > 0:
-        print("[%s] Loading data from %s" % (
-            class_name,
-            _utils._repr_niimgs(imgs)[:200]))
-
-    # Get series from data with optional smoothing
-    if verbose > 1:
-        print("[%s] Masking and smoothing" % class_name)
-    data = masking.apply_mask(imgs, mask_img_,
-                              smoothing_fwhm=parameters['smoothing_fwhm'])
+        print("[%s] Extracting region signals" % class_name)
+    region_signals, aux = cache(extraction_function, memory,
+                                func_memory_level=2,
+                                memory_level=memory_level)(imgs)
 
```

Comment:
I am wondering about the change in the association between smoothing and masking: prior to this PR, I understand that no smoothing was performed with signals outside of the mask, while now, the images are smoothed irrespective of the mask. This may have consequences when the mask is used to avoid some noise sources (artefacted voxels, vessels etc.)